siad/daemon.go

```diff
@@ -107,13 +108,36 @@ func createSubdirs(rootDir string) error {
 }
 
 // TODO: move this to the state module?
-func (d *daemon) relayBlock(conn modules.NetConn) error {
+func (d *daemon) acceptBlock(conn modules.NetConn) error {
 	var b consensus.Block
 	err := conn.ReadObject(&b, consensus.BlockSizeLimit)
 	if err != nil {
 		return err
 	}
-	return d.gateway.RelayBlock(b)
+
+	err = d.state.AcceptBlock(b)
+	if err == consensus.ErrOrphan {
+		go d.gateway.Synchronize(conn.Addr())
```

Comment:
If calling Synchronize always returns an orphan (because that peer is being malicious), will this cause problems? Or will the sillyness stop after 1 call?